### ISSUE TITLE
Possible connections leak

### ISSUE THREAD

**Issue by FDoKE** [tarantool/cartridge-java#187]
Looks like leak present in the client.
Our use case
1. Creating new client
`new ClusterTarantoolTupleClient(config, Collections.singletonList(routerAddress))`
3. Getting list of another tarantool servers
4. Closing this client
5. Opening new client to all that we got in 2. 
`new ProxyTarantoolTupleClient(new ClusterTarantoolTupleClient(config, discoveryClientProvider))`
6. doing our work
7. Closing this client

And looks like there is connection left after closing all clients.

**Connections begins to increment with each new client and not freed after closing until system return 'Too many files'**
```
sudo lsof -u pxf | grep -ic tarantool-host
1255
```
this repeats in result:
```
java    11592  pxf  370u     IPv4          555139027       0t0       TCP client-host:19489->tarantool-host:mcs-fastmail (ESTABLISHED)
java    11592  pxf  419u     IPv4          555139028       0t0       TCP client-host:14578->tarantool-host:odette-ftp (ESTABLISHED)
java    11592  pxf  420u     IPv4          555141070       0t0       TCP client-host:19433->tarantool-host:mcs-fastmail (ESTABLISHED)
java    11592  pxf  421u     IPv4          555141256       0t0       TCP client-host:14522->tarantool-host:odette-ftp (ESTABLISHED)
java    11592  pxf  422u     IPv4          555141257       0t0       TCP client-host:23014->tarantool-host:3301 (ESTABLISHED)
java    11592  pxf  423u     IPv4          555142295       0t0       TCP client-host:33345->tarantool-host:mysql (ESTABLISHED)
java    11592  pxf  424u     IPv4          555141071       0t0       TCP client-host:19447->tarantool-host:mcs-fastmail (ESTABLISHED)
java    11592  pxf  425u     IPv4          555138865       0t0       TCP client-host:19437->tarantool-host:mcs-fastmail (ESTABLISHED)
java    11592  pxf  426u     IPv4          555141072       0t0       TCP client-host:14554->tarantool-host:odette-ftp (ESTABLISHED)
java    11592  pxf  427u     IPv4          555142296       0t0       TCP client-host:19459->tarantool-host:mcs-fastmail (ESTABLISHED)
java    11592  pxf  428u     IPv4          555142297       0t0       TCP client-host:59359->tarantool-host:opsession-prxy (ESTABLISHED)
java    11592  pxf  429u     IPv4          555141258       0t0       TCP client-host:14542->tarantool-host:odette-ftp (ESTABLISHED)
```

Link to our product: https://github.com/datamarts/tarantool-pxf-connector
**I tried new 0.7.0 version of cartridge java, probem still persists** (not open sourced yet)

**Comment by vrogach2020** [comment]
Looks like another case of https://github.com/tarantool/cartridge-java/pull/183
@wey1and  we need to check why connections do not autoclose when client is closed.

**Comment by akudiyar** [comment]
It seems to be fixed in https://github.com/tarantool/cartridge-java/commit/dcd054082ac1737c9b174b4636fcb409e0504543.
@FDoKE would you mind building the driver package form master and checking it again in your environment?

**Comment by FDoKE** [comment]
@akudiyar checked master, looks good, problem solved! 👍 
Can you tell when this fix will be released? Would be good if asap

**Comment by vrogach2020** [comment]
@FDoKE we are going to release asap. The deadline for this release comes in 2 weeks.  

@wey1and please double check the case when the client is closed or garbage collected

**Comment by wey1and** [comment]
@vrogach2020 I tried on my computer. When the client is closed, the connections are closed immediately. I think we can close this issue.

**Comment by FDoKE** [comment]
@wey1and looks like after some time problem reproduces itself (worked well for some time)
```
sudo lsof -u pxf | grep -ic tarantool-host
4935
```

**Comment by vrogach2020** [comment]
Fixed in 0.7.1
